stl/inc/chrono

```diff
@@ -3472,66 +3472,49 @@ namespace chrono {
+            // A distinction has to be made here between clocks that always count 86,400 seconds/day (system_clock,
+            // tai_clock, and gps_clock) and ones that count leap seconds (utc_clock and, since 2018-07-01, file_clock).
+            // The correct tick count for the clocks that always count 86,400 seconds/day can be determined without
             // knowing whether any leap seconds have occurred, and there aren't any invalid times due to negative leap
             // second deletions.
-            //
-            // system_clock also has 86,400 seconds/day, but observes negative leap seconds by skipping them in its tick
-            // count. So leap second data is needed to check for a valid time, but not to calculate the tick count.
 
-            if (_Leap != _LeapSecondRep::_None) {
-                if (_Hour_24 == 23 && _Minute == 59 && _Second >= 59) {
+            if (_Leap == _LeapSecondRep::_All) {
+                constexpr bool _Can_rep_sec = _Can_represent<_DurationType, seconds>();
+                if constexpr (_Can_rep_sec) {
                     // It's possible that the parsed time doesn't exist because (a) _Seconds == 60 and there *isn't* a
                     // leap second insertion or (b) _Seconds == 59 and there *is* a leap second subtraction.
-
-                    // Check for quick exit.
                     const auto& _Tzdb{_CHRONO get_tzdb()};
-                    if (_Leap == _LeapSecondRep::_Negative && _Tzdb._All_ls_positive) {
-                        return true;
-                    }
 
                     const bool _Possible_insertion{_Second == 60};
                     const sys_seconds _Target_sys_time{
-                        _CHRONO floor<seconds>(_Dur) + (_Possible_insertion ? seconds{0} : seconds{1})};
+                        _CHRONO floor<seconds>(_Dur) - (_Possible_insertion ? seconds{1} : seconds{0})};
                     const auto& _Ls_vector{_Tzdb.leap_seconds};
-                    const auto _It{_STD lower_bound(_Ls_vector.begin(), _Ls_vector.end(), _Target_sys_time)};
-                    const bool _Match_leap{_It != _Ls_vector.end() && *_It == _Target_sys_time};
+                    const auto _It{_STD upper_bound(_Ls_vector.begin(), _Ls_vector.end(), _Target_sys_time)};
+                    const bool _Match_leap{_It != _Ls_vector.end() && *_It == _Target_sys_time + seconds{1}};
 
-                    // Condition for a good parse: (_Seconds == 59 && !_Match_leap) || (_Match_leap &&
-                    // _It->_Is_positive()). Below is the inverse of this.
-                    if (!(_Match_leap ? _It->_Positive() : !_Possible_insertion)) {
+                    // Condition for a good parse: (_Second < 60 && !_Match_leap) || (_Match_leap && _It->_Positive())
+                    // || (_Second < 59 && _Match_leap && !_It->_Positive()). Below is the inverse of this.
+                    if (!(_Match_leap ? _It->_Positive() || _Second < 59 : !_Possible_insertion)) {

```

Comment:
oh boy I do not like using a ternary as part of an if condition in this way (it was bad before, but not as bad). I would much prefer just using the actual conditions, even if they are slightly longer.

Reply:
This is technically pre-existing, but you made it worse since now there's another arm of logic inside the ternary.